### ISSUE TITLE
Can not run the task 2 with codecept support appium v2

### ISSUE THREAD

**Issue by xuanmai-agilityio** [kobenguyent/thanh-nguyen#1]
Hi there, really appreciate your demo for the Appium v2 compatibility. I'm quite new with the automation test and Codeceptjs specifically.
I've tried to pull the repo and set it up to run the test but it didn't work.
### Step to reproduce

1. Pull the repo and install dependencies following the instructions in task 2.
2. Patch the Codeceptjs helper following the [#3622](https://github.com/codeceptjs/CodeceptJS/pull/3622/files)
3. Run ```appium``` to start the Appium server.
4. Run ```yarn android-tests``` to start testing.

### Issue
- The test failed with these errors
<img width="593" alt="Screen Shot 2023-05-07 at 16 27 21" src="https://user-images.githubusercontent.com/61130765/236669303-648e4ea4-8174-42e4-a4a6-3f9360374ccc.png">

### Expected behavior
- The test should run properly with the passed result.

It would be great if you take a look at this issue and give me some hints to solve this. Thank you in advance!

**Comment by kobenguyent** [comment]
Hey @xuanmai-agilityio may you share if you changed something of the configuration?

**Comment by xuanmai-agilityio** [comment]
I didn't change the `codecept.conf.ts` file but had some changes in the Android caps.
<img width="510" alt="Screen Shot 2023-05-07 at 17 51 58" src="https://user-images.githubusercontent.com/61130765/236673205-c4cb990b-15cc-4d2f-886a-109a8571e8e2.png">
Because my Android emulator is `Pixel_5_API_30` and run on version `13.0`
<img width="445" alt="Screen Shot 2023-05-07 at 17 54 03" src="https://user-images.githubusercontent.com/61130765/236673228-347d2a4e-c11e-4e2a-b634-7a788af68567.png">
I also patched the codeceptjs following the PR above using `patch-package`. The patch is for `Appium.js` and `package.json` files. Did I do it correctly?

**Comment by kobenguyent** [comment]
Thanks for that! May you give a try installing the branch directly? 

@xuanmai-agilityio I found the issue and patched it. 

```
th~$npm run start-appium

> monefy-codeceptjs@1.0.0 start-appium
> appium --base-path=/wd/hub

[Appium] Welcome to Appium v2.0.0-beta.57 (REV 9cd9fcfd49e3dfc294fdbc5f6ba9f899a002ee76)
[Appium] Non-default server args:
[Appium] {
[Appium]   basePath: '/wd/hub'
[Appium] }
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at /Users/trung-thanh/Desktop/thanh-nguyen/task2/node_modules/appium-espresso-driver
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at /Users/trung-thanh/Desktop/thanh-nguyen/task2/node_modules/appium-uiautomator2-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at /Users/trung-thanh/Desktop/thanh-nguyen/task2/node_modules/appium-xcuitest-driver
(node:3236) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on 0.0.0.0:4723

```

It works for me without any issue. 

```
tth~$npm run android-tests

> monefy-codeceptjs@1.0.0 android-tests
> export PLATFORM=android && codeceptjs run --verbose --grep @android

CodeceptJS v3.4.1 #StandWithUkraine
Using test root "/Users/trung-thanh/Desktop/thanh-nguyen/task2"
Helpers: Appium
Plugins: screenshotOnFail

Balance @android --
    [1]  Starting recording promises
    Timeouts: 
  Adding income successfully
    androidHomeScreen: addNewIncome "50"
      I wait 2
      I tap "#com.monefy.app.pro:id/income_button"
      I wait 2
      I tap "#com.monefy.app.pro:id/buttonKeyboard5"
      I wait 2
      I tap "#com.monefy.app.pro:id/buttonKeyboard0"
      I wait 2
      I tap "#com.monefy.app.pro:id/keyboard_action_button"
      I wait 2
      I tap "#com.monefy.app.pro:id/imageView"
    I: waitAndSeeText "50"
      I wait 2
      I see "50"
  ✔ OK in 17018ms


  OK  | 1 passed   // 22s

```

**Comment by xuanmai-agilityio** [comment]
Many thanks @kobenguyent , I've tried installing the branch directly and it worked. Btw, could you help to share settings for the iOS capabilities? Are there any special values that are worth noticing?

**Comment by kobenguyent** [comment]
hey @xuanmai-agilityio I guess for iOS pretty much nothing special to be aware. Unfortunately, I don't have the test iOS app to try so can't give you the example.

**Comment by xuanmai-agilityio** [comment]
@kobenguyent  I also found another issue with the `_startBrowser` function, when appium run the 2nd test case, it will run this function and it will convert the existing options and one more `appium:` prefix in the config. So I think we should only convert it one time when the Appium starts.
So I've fixed it as below and it runs smoothly.
<img width="607" alt="Screen Shot 2023-05-09 at 10 49 59" src="https://user-images.githubusercontent.com/61130765/236989332-9e5390c7-b6d8-4629-ae37-4c19e193274c.png">
